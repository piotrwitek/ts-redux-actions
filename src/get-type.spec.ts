import { createAction, getType } from '.';

describe('getType', () => {

  // TODO: #3
  // should error when missing argument
  // should error when passed invalid arguments
  // check object, empty array, primitives

  it('no payload', () => {
    const increment = createAction('INCREMENT');

    const type: 'INCREMENT' = getType(increment);
    expect(type).toBe('INCREMENT');
  });

  it('no payload alternative', () => {
    const increment = createAction('INCREMENT', () => ({ type: 'INCREMENT' }));

    const type: 'INCREMENT' = getType(increment);
    expect(type).toBe('INCREMENT');
  });

  it('with payload', () => {
    const add = createAction('ADD',
      (amount: number) => ({ type: 'ADD', payload: amount }),
    );

    const type: 'ADD' = getType(add);
    expect(type).toBe('ADD');
  });

  it('with payload and meta', () => {
    const notify = createAction('NOTIFY',
      (username: string, message: string) => ({
        type: 'NOTIFY',
        payload: { message: `${username}: ${message}` },
        meta: { username, message },
      }),
    );

    const type: 'NOTIFY' = getType(notify);
    expect(type).toBe('NOTIFY');
  });

  it('with payload and no params', () => {
    const showNotification = createAction('SHOW_NOTIFICATION',
      () => ({
        type: 'SHOW_NOTIFICATION',
        payload: 'default message',
      }),
    );

    const type: 'SHOW_NOTIFICATION' = getType(showNotification);
    expect(type).toBe('SHOW_NOTIFICATION');
  });

  it('with payload and optional param', () => {
    const showNotification = createAction('SHOW_NOTIFICATION',
      (message?: string) => ({
        type: 'SHOW_NOTIFICATION',
        payload: message,
      }),
    );

    const type: 'SHOW_NOTIFICATION' = getType(showNotification);
    expect(type).toBe('SHOW_NOTIFICATION');
  });

  it('with meta and no params', () => {
    const showError = createAction('SHOW_ERROR',
      () => ({
        type: 'SHOW_ERROR',
        meta: { type: 'error' },
      }),
    );

    const type: 'SHOW_ERROR' = getType(showError);
    expect(type).toBe('SHOW_ERROR');
  });

  it('with meta and optional param', () => {
    const showError = createAction('SHOW_ERROR',
      (message?: string) => ({
        type: 'SHOW_ERROR',
        payload: message,
        meta: { type: 'error' },
      }),
    );

    const type: 'SHOW_ERROR' = getType(showError);
    expect(type).toBe('SHOW_ERROR');
  });

});
