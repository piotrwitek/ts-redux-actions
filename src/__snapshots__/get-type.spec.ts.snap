// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getType(createAction('CREATE_ACTION')) 1`] = `"string"`;

exports[`getType(createCustomAction('CREATE_CUSTOM_ACTION')) 1`] = `"string"`;

exports[`getType(createStandardAction('CREATE_STANDARD_ACTION')<void>()) 1`] = `"\\"CREATE_STANDARD_ACTION\\""`;
